nexus/go-handler.md

```diff
@@ -0,0 +1,307 @@
+# Temporal Go SDK - Nexus Handler
+
+## Overview
+
+An addition to the [Temporal Go SDK][temporal-go-sdk] that exposes the ability for users to handle Nexus API requests.
+
+**Goals**
+
+- Expose a simple, high level API for binding Nexus operations to common Temporal constructs
+- Support for starting and canceling operations
+- Support multi-tenant applications with minimal boilerplate
+
+**Non-Goals**
+
+- Support for getting an operation's result or info
+- API definitions for other SDKs
+- API for invoking Nexus calls - refer to the [Nexus Go SDK][nexus-go-sdk] for an out-of-workflow reference client
+- Server implementation details
+- Worker / server API contract
+- Allow advanced users to implement Nexus inbound and outbound request proxies for cross-organization calls where
+  data converters are not shared and data is not exposed to Temporal server.
+
+  This will quickly become a requirement for production use cases, especially with Temporal Cloud's zero trust model. It
+  will be addressed shortly after Nexus becomes available for preview.
+
+> NOTE: These are non-goals for _this_ proposal, but they will be defined in follow up proposals / PRs.
+
+## Background
+
+Nexus is a thin protocol on top of HTTP, defined in [this spec][nexus-http-api]. Temporal serves as an HTTP proxy,
+translating Nexus requests into tasks that can be polled on by an SDK worker on its configured task queue.
+Due to the synchronous nature of HTTP, the generated tasks are not persisted, and require an available worker
+to be processed.
```

Comment:
In an async start operation,  before we can return a "201 created" we have to make sure that either the Nexus request+id is persisted, or the generated tasks+id are persisted. Otherwise, we force a stateless client to do retries, and that defeats the purpose of nexus imho...

Reply:
They'll be persisted if the client gets a response.

Reply:
That's great, we need to make that explicit in the text, this is important...

Reply:
+1, I don't understand this sentence:

> Due to the synchronous nature of HTTP, the generated tasks are not persisted, and require an available worker to be processed.

My understanding is that the whole point of Nexus is that it is adding the possibility for asynchronicity to HTTP. But I read the sentence as saying that it only supports synchronous responses.